src/BenchmarkDotNet/Running/BenchmarkSwitcher.cs

```diff
@@ -72,15 +73,28 @@ public class BenchmarkSwitcher
                 return Enumerable.Empty<Summary>();
             }
 
-            var globalChronometer = Chronometer.Start();
-            var summaries = new List<Summary>();
-
             var effectiveConfig = ManualConfig.Union(config ?? DefaultConfig.Instance, parsedConfig);
 
-            var filteredBenchmarks = typeParser.Filter(effectiveConfig);
+            var listBenchmarkCase = options.ListBenchmarkCaseMode != ListBenchmarkCaseMode.Disable;
+
+            var filteredBenchmarks = typeParser.Filter(effectiveConfig, listBenchmarkCase);
             if (filteredBenchmarks.IsEmpty())
                 return Array.Empty<Summary>();
 
+            if (listBenchmarkCase)
+            {
+                var testName = filteredBenchmarks.SelectMany(p => p.BenchmarksCases)
+                    .Select(p => p.Descriptor.Type.Namespace + "." + p.Descriptor.DisplayInfo).Distinct();
```

Comment:
the benchmarks are filtered using following code:

https://github.com/dotnet/BenchmarkDotNet/blob/1e62355f209a25c7a33f9ab7e7e03b0afe7d851f/src/BenchmarkDotNet/Filters/GlobFilter.cs#L22

could you please use `benchmarkCase.Descriptor.GetFilterName()` instead of `p.Descriptor.Type.Namespace + "." + p.Descriptor.DisplayInfo` ?